api/python/quilt3/data_transfer.py

```diff
@@ -277,6 +400,8 @@ def _copy_file_list_internal(s3_client, file_list, callback):
 
     stopped = False
 
+    s3_client_provider = S3ClientProvider()  # Share provider accross threads to reduce redundant public bucket checks
```

Comment:
Then it needs to be thread-safe?

Reply:
It is threadsafe for this usage AFAICT. Logically it is a cache representation of a static dataset, so the concurrency requirements are very simple